daphne_worker/src/dap.rs

```diff
@@ -531,28 +534,31 @@ impl<D> DapHelper<BearerToken> for DaphneWorkerConfig<D> {
     ) -> std::result::Result<HashMap<Nonce, TransitionFailure>, DapError> {
         let task_config = self.try_get_task_config_for(task_id)?;
 
-        // TODO Coalesce nonces with the same batch name into a single DO request.
-        let mut early_fails = HashMap::new();
         let namespace = self.durable_object(BINDING_DAP_REPORT_STORE)?;
+        let mut durable_requests: HashMap<String, Vec<String>> = HashMap::new();
         for report_share in report_shares.iter() {
             let durable_name =
                 self.durable_report_store_name(task_config, task_id, &report_share.nonce);
+            let nonce_hex = hex::encode(report_share.nonce.get_encoded());
+            if let Some(nonce_hex_set) = durable_requests.get_mut(&durable_name) {
+                nonce_hex_set.push(nonce_hex);
+            } else {
+                durable_requests.insert(durable_name, vec![nonce_hex]);
+            }
+        }
 
-            // TODO Don't block on DO requests (issue multiple requests simultaneously).
+        // TODO Run requests in parallel.
```

Comment:
Does this mean run the POSTs to each `durable_name` in parallel? (Just want to clarify)

Reply:
No, this change just coalesces requests to each distinct DO isntance. The requests themselves are still serial. Note that this gets fixed higher in the stack.